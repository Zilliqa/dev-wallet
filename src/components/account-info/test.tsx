/**
 * This file is part of nucleus-wallet.
 * Copyright (c) 2018 - present Zilliqa Research Pte. Ltd.
 *
 * nucleus-wallet is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * nucleus-wallet is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * nucleus-wallet.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render, wait } from '@testing-library/react';
import AccountInfo from '.';

test('matches the snapshot when loaded', async () => {
  const getBalance = jest.fn().mockResolvedValue('100000');
  const address = '0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0';
  const { container, getByTestId } = render(
    <AccountInfo address={address} getBalance={getBalance} />
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('matches the snapshot when loaded', async () => {
  const getBalance = jest.fn().mockResolvedValue('100000');
  const address = '0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0';
  const { container, getByTestId } = render(
    <AccountInfo address={address} getBalance={getBalance} />
  );

  await wait(() => getByTestId('container-data'));
  expect(container.firstChild).toMatchSnapshot();
});

test('matches the snapshot when failed', async () => {
  const getBalance = jest.fn().mockRejectedValue('Intended Error');
  const address = '0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0';
  const { container, getByTestId } = render(
    <AccountInfo address={address} getBalance={getBalance} />
  );
  await wait(() => getByTestId('container-error'));
  expect(container.firstChild).toMatchSnapshot();
});

test('matches the snapshot when no data', async () => {
  const getBalance = jest.fn().mockResolvedValue(undefined);
  const address = '0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0';
  const { container, getByTestId } = render(
    <AccountInfo address={address} getBalance={getBalance} />
  );
  await wait(() => getByTestId('container-no-data'));
  expect(container.firstChild).toMatchSnapshot();
});
