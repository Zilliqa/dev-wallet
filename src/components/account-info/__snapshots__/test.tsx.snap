// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when failed 1`] = `
.c0 {
  outline: none;
  margin: 0;
  font-family: inherit;
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0::-webkit-file-upload-button {
  font: inherit;
  -webkit-appearance: button;
}

.c0 [type='button'],
.c0 [type='reset'],
.c0 [type='submit'] {
  -webkit-appearance: button;
}

.c0:not(:disabled),
.c0 [type='button']:not(:disabled),
.c0 [type='reset']:not(:disabled),
.c0 [type='submit']:not(:disabled) {
  cursor: pointer;
}

.c0::-moz-focus-inner,
.c0 [type='button']::-moz-focus-inner,
.c0 [type='reset']::-moz-focus-inner,
.c0 [type='submit']::-moz-focus-inner {
  padding: 0;
  border-style: none;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 {
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
}

.c0 {
  background-color: transparent;
  color: #585968;
  border-color: transparent;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #242536;
}

.c0:disabled {
  color: #A0A1B2;
}

<div>
  <div
    class="px-4"
  >
    <h5>
      <b>
        Account Info
      </b>
    </h5>
    <div
      class="d-flex"
    >
      <div
        class="py-2"
      >
        <div
          class="paper"
          style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(3, 85, 94); height: 100px; width: 100px;"
        >
          <svg
            height="100"
            width="100"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <rect
              fill="#FA6800"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.332176576371118 5.932298588714297) rotate(79.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#236FE1"
              height="100"
              rx="0"
              ry="0"
              transform="translate(57.127223703787955 -31.273129535154165) rotate(409.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#C81456"
              height="100"
              rx="0"
              ry="0"
              transform="translate(56.64908831248396 43.86654244724025) rotate(76.8 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#FB186B"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.595406197345499 -101.65110911481909) rotate(394.6 50 50)"
              width="100"
              x="0"
              y="0"
            />
          </svg>
        </div>
      </div>
      <div
        class="px-4 text-left text-secondary"
      >
        <b>
          Address
        </b>
        <p
          class="pt-1 font-monospace"
        >
          <a
            href="https://viewblock.io/zilliqa/address/zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v?network=testnet"
            rel="noopener noreferrer"
            target="_blank"
          >
            zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v
          </a>
           
          <br />
          <small>
            (ByStr20: 0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0)
          </small>
        </p>
        <b>
          Balance
          <button
            aria-label=""
            class="mb-1 py-0 px-1 c0"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
             
            
             
          </button>
        </b>
        <div
          data-testid="container-error"
        >
          Something went wrong: undefined
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the snapshot when loaded 1`] = `
.c0 {
  outline: none;
  margin: 0;
  font-family: inherit;
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0::-webkit-file-upload-button {
  font: inherit;
  -webkit-appearance: button;
}

.c0 [type='button'],
.c0 [type='reset'],
.c0 [type='submit'] {
  -webkit-appearance: button;
}

.c0:not(:disabled),
.c0 [type='button']:not(:disabled),
.c0 [type='reset']:not(:disabled),
.c0 [type='submit']:not(:disabled) {
  cursor: pointer;
}

.c0::-moz-focus-inner,
.c0 [type='button']::-moz-focus-inner,
.c0 [type='reset']::-moz-focus-inner,
.c0 [type='submit']::-moz-focus-inner {
  padding: 0;
  border-style: none;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 {
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
}

.c0 {
  background-color: transparent;
  color: #585968;
  border-color: transparent;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #242536;
}

.c0:disabled {
  color: #A0A1B2;
}

<div>
  <div
    class="px-4"
  >
    <h5>
      <b>
        Account Info
      </b>
    </h5>
    <div
      class="d-flex"
    >
      <div
        class="py-2"
      >
        <div
          class="paper"
          style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(3, 85, 94); height: 100px; width: 100px;"
        >
          <svg
            height="100"
            width="100"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <rect
              fill="#FA6800"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.332176576371118 5.932298588714297) rotate(79.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#236FE1"
              height="100"
              rx="0"
              ry="0"
              transform="translate(57.127223703787955 -31.273129535154165) rotate(409.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#C81456"
              height="100"
              rx="0"
              ry="0"
              transform="translate(56.64908831248396 43.86654244724025) rotate(76.8 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#FB186B"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.595406197345499 -101.65110911481909) rotate(394.6 50 50)"
              width="100"
              x="0"
              y="0"
            />
          </svg>
        </div>
      </div>
      <div
        class="px-4 text-left text-secondary"
      >
        <b>
          Address
        </b>
        <p
          class="pt-1 font-monospace"
        >
          <a
            href="https://viewblock.io/zilliqa/address/zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v?network=testnet"
            rel="noopener noreferrer"
            target="_blank"
          >
            zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v
          </a>
           
          <br />
          <small>
            (ByStr20: 0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0)
          </small>
        </p>
        <b>
          Balance
          <button
            aria-label=""
            class="mb-1 py-0 px-1 c0"
            disabled=""
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
             
            
             
          </button>
        </b>
        <div
          data-testid="container-loading"
        >
          <small>
            Loading...
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the snapshot when loaded 2`] = `
.c0 {
  outline: none;
  margin: 0;
  font-family: inherit;
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0::-webkit-file-upload-button {
  font: inherit;
  -webkit-appearance: button;
}

.c0 [type='button'],
.c0 [type='reset'],
.c0 [type='submit'] {
  -webkit-appearance: button;
}

.c0:not(:disabled),
.c0 [type='button']:not(:disabled),
.c0 [type='reset']:not(:disabled),
.c0 [type='submit']:not(:disabled) {
  cursor: pointer;
}

.c0::-moz-focus-inner,
.c0 [type='button']::-moz-focus-inner,
.c0 [type='reset']::-moz-focus-inner,
.c0 [type='submit']::-moz-focus-inner {
  padding: 0;
  border-style: none;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 {
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
}

.c0 {
  background-color: transparent;
  color: #585968;
  border-color: transparent;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #242536;
}

.c0:disabled {
  color: #A0A1B2;
}

<div>
  <div
    class="px-4"
  >
    <h5>
      <b>
        Account Info
      </b>
    </h5>
    <div
      class="d-flex"
    >
      <div
        class="py-2"
      >
        <div
          class="paper"
          style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(3, 85, 94); height: 100px; width: 100px;"
        >
          <svg
            height="100"
            width="100"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <rect
              fill="#FA6800"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.332176576371118 5.932298588714297) rotate(79.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#236FE1"
              height="100"
              rx="0"
              ry="0"
              transform="translate(57.127223703787955 -31.273129535154165) rotate(409.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#C81456"
              height="100"
              rx="0"
              ry="0"
              transform="translate(56.64908831248396 43.86654244724025) rotate(76.8 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#FB186B"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.595406197345499 -101.65110911481909) rotate(394.6 50 50)"
              width="100"
              x="0"
              y="0"
            />
          </svg>
        </div>
      </div>
      <div
        class="px-4 text-left text-secondary"
      >
        <b>
          Address
        </b>
        <p
          class="pt-1 font-monospace"
        >
          <a
            href="https://viewblock.io/zilliqa/address/zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v?network=testnet"
            rel="noopener noreferrer"
            target="_blank"
          >
            zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v
          </a>
           
          <br />
          <small>
            (ByStr20: 0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0)
          </small>
        </p>
        <b>
          Balance
          <button
            aria-label=""
            class="mb-1 py-0 px-1 c0"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
             
            
             
          </button>
        </b>
        <div
          data-testid="container-data"
        >
          <small>
            0.0000001 ZIL
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the snapshot when no data 1`] = `
.c0 {
  outline: none;
  margin: 0;
  font-family: inherit;
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c0::-webkit-file-upload-button {
  font: inherit;
  -webkit-appearance: button;
}

.c0 [type='button'],
.c0 [type='reset'],
.c0 [type='submit'] {
  -webkit-appearance: button;
}

.c0:not(:disabled),
.c0 [type='button']:not(:disabled),
.c0 [type='reset']:not(:disabled),
.c0 [type='submit']:not(:disabled) {
  cursor: pointer;
}

.c0::-moz-focus-inner,
.c0 [type='button']::-moz-focus-inner,
.c0 [type='reset']::-moz-focus-inner,
.c0 [type='submit']::-moz-focus-inner {
  padding: 0;
  border-style: none;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 {
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
}

.c0 {
  background-color: transparent;
  color: #585968;
  border-color: transparent;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #242536;
}

.c0:disabled {
  color: #A0A1B2;
}

<div>
  <div
    class="px-4"
  >
    <h5>
      <b>
        Account Info
      </b>
    </h5>
    <div
      class="d-flex"
    >
      <div
        class="py-2"
      >
        <div
          class="paper"
          style="border-radius: 50px; display: inline-block; margin: 0px; overflow: hidden; padding: 0px; background-color: rgb(3, 85, 94); height: 100px; width: 100px;"
        >
          <svg
            height="100"
            width="100"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <rect
              fill="#FA6800"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.332176576371118 5.932298588714297) rotate(79.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#236FE1"
              height="100"
              rx="0"
              ry="0"
              transform="translate(57.127223703787955 -31.273129535154165) rotate(409.1 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#C81456"
              height="100"
              rx="0"
              ry="0"
              transform="translate(56.64908831248396 43.86654244724025) rotate(76.8 50 50)"
              width="100"
              x="0"
              y="0"
            />
            <rect
              fill="#FB186B"
              height="100"
              rx="0"
              ry="0"
              transform="translate(7.595406197345499 -101.65110911481909) rotate(394.6 50 50)"
              width="100"
              x="0"
              y="0"
            />
          </svg>
        </div>
      </div>
      <div
        class="px-4 text-left text-secondary"
      >
        <b>
          Address
        </b>
        <p
          class="pt-1 font-monospace"
        >
          <a
            href="https://viewblock.io/zilliqa/address/zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v?network=testnet"
            rel="noopener noreferrer"
            target="_blank"
          >
            zil18j0lvshp0t6ucrqsjkfunpjdg55my3aqenns9v
          </a>
           
          <br />
          <small>
            (ByStr20: 0x3C9Ff642E17aF5Cc0C109593C9864d4529B247A0)
          </small>
        </p>
        <b>
          Balance
          <button
            aria-label=""
            class="mb-1 py-0 px-1 c0"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
             
            
             
          </button>
        </b>
        <div
          data-testid="container-no-data"
        >
          No data
        </div>
      </div>
    </div>
  </div>
</div>
`;
